### ISSUE TITLE
Backing property in a companion object triggers a backing-property-naming violation

### ISSUE THREAD

**Issue by ygnessin** [pinterest/ktlint#2877]
Thank you so much for maintaining this library and for your attention to this issue.

Is this a bug, or is it intended behavior? If intended, I can open a feature request, because I don't think this should be a violation.

## Expected Behavior

no ktlint violations

## Observed Behavior

```
Lint error > [standard:backing-property-naming] Backing property is only allowed when a matching property or function exists
```

## Steps to Reproduce

```kotlin
    companion object {
        private var _isEnabled = false
    }

    val isEnabled: Boolean
        get() = _isEnabled
```

## Your Environment

* Version of ktlint used: 1.4.1
* Relevant parts of the `.editorconfig` settings: N/A
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): https://github.com/jeremymailen/kotlinter-gradle 4.5.0
* Version of Gradle used (if applicable): 8.11.1
* Operating System and version: macOS 14.7.1

**Comment by paul-dingemans** [comment]
I cannot find and reference/definition/example on whether a backing property may be declared in the companion object. So it seems valid to allow this. No separate issue is needed for that.

**Comment by paul-dingemans** [comment]
Actually just found a [reference](https://kotlinlang.org/docs/java-to-kotlin-interop.html):
> Kotlin properties declared in a named object or a companion object will have static backing fields either in that named object or in the class containing the companion object.

**Comment by paul-dingemans** [comment]
Yes, but it at least it confirms the idea that a backing property can be defined in the companion object.
